code/modules/food_and_drinks/kitchen_machinery/food_replicator.dm

```diff
@@ -0,0 +1,631 @@
+/obj/machinery/food_replicator/proc/adjust_paste(amount = 0)
+	food_paste = max(0, (food_paste + amount))
+
+	if(food_paste > max_food_paste)
+		var/paste_to_waste = food_paste - max_food_paste
+		if(paste_to_waste >= 20)
+			print_poly(20, 1)
+		else if(paste_to_waste >= 12)
+			print_poly(12, 1)
+		else if(paste_to_waste >= 8)
+			print_poly(8, 1)
+		else if(paste_to_waste >= 6)
+			print_poly(6, 1)
+		else if(paste_to_waste >= 4)
+			print_poly(4, 1)
+		else
+			print_poly(1, 1)
+
+/obj/machinery/food_replicator/attack_ai(mob/user)
+	return attack_hand(user)
+
+/obj/machinery/food_replicator/attack_hand(mob/user)
+	if(stat & (BROKEN|NOPOWER))
+		return
+	user.set_machine(src)
+	ui_interact(user)
+
+/obj/machinery/food_replicator/interact(mob/user)
+	if(panel_open || !anchored)
+		return
+	//todo
+
+/obj/machinery/food_replicator/attackby(obj/item/O, mob/user, params)
+	if(operating)
+		return
+	if(default_deconstruction_screwdriver(user, open_icon, off_icon, O))
+		return
+	if(default_unfasten_wrench(user, O))
+		return
+	if(exchange_parts(user, O))
+		return
+	default_deconstruction_crowbar(O)
+
+	if(istype(O, /obj/item/weapon/reagent_containers/food/snacks))
+		if(loaded)
+			to_chat(user, "<span class='warning'>There is already a dish loaded. Recycle or eject it before adding a new dish.</span>")
+			return
+		var/obj/item/weapon/reagent_containers/food/snacks/to_load = O
+		if(!user.canUnEquip(to_load, 0))
+			to_chat(user, "<span class='warning'>[to_load] is stuck to your hand, you can't seem to put it down!</span>")
+			return
+		user.unEquip(to_load)
+		to_load.forceMove(src)
+		loaded = to_load
+		nanomanager.update_uis(src)
+		return 1
+
+	if(panel_open && iswire(O))
+		var/obj/item/stack/cable_coil/C = O
+		if(stat & BROKEN)
+			if(C.amount >= 5)
+				if(do_after(user, 40, target = src))
+					to_chat(user, "<span class='notice'>You replace the burnt out wiring, restoring functionality.</span>")
+					C.use(5)
+					set_fix()
+			else
+				to_chat(user, "<span class='warning'>You don't have enough cable to repair the wiring.</span>")
+		else
+			to_chat(user, "<span class='notice'>The wiring doesn't require replacing at this time.</span>")
+		return 1
+
+	if(istype(O, /obj/item/weapon/card/emag))
+		emag_act(user)
+		return 1
+
+/obj/machinery/food_replicator/emag_act(mob/user)
+	if(!emagged)
+		to_chat(user, "<span class='warning'>You corrupt the reagent replication subroutine to lace created dishes with poisons.</span>")
+		emagged = 1
+	else
+		to_chat(user, "<span class='warning'>The reagent replication subroutines are already corrupted!</span>")
+
+/obj/machinery/food_replicator/proc/set_broken()
+	stat |= BROKEN
+	//make a shower of sparks to communicate the wires burned out
+	var/datum/effect/system/spark_spread/s = new /datum/effect/system/spark_spread
+	s.set_up(2, 1, src)
+	s.start()
+	use_power = 0	//while we are busted, we won't draw power
```

Comment:
According to the comments beginning at https://github.com/ParadiseSS13/Paradise/blob/master/code/game/machinery/machinery.dm#L37 , there should be flags that might be better suited for various states, but they don't seem to be used at a quick glance through that file, so perhaps that could be something we address in a separate PR.